miio/huizuo.py

```diff
@@ -147,7 +311,288 @@ def set_brightness(self, level):
+
+    @command(
+        default_output=format_output("Fan powering on"),
+    )
+    def fan_on(self):
+        """Power fan on (only for models with fan)."""
+        if self.model in MODELS_WITH_FAN_WY or self.model in MODELS_WITH_FAN_WY2:
+            return self.set_property("fan_power", True)
+        else:
+            raise HuizuoException("Your device doesn't support a fan management")
+
+    @command(
+        default_output=format_output("Fan powering off"),
+    )
+    def fan_off(self):
+        """Power fan off (only for models with fan)."""
+        if self.model in MODELS_WITH_FAN_WY or self.model in MODELS_WITH_FAN_WY2:
+            return self.set_property("fan_power", False)
+        else:
+            raise HuizuoException("Your device doesn't support a fan management")
+
+    @command(
+        click.argument("fan_level", type=int),
+        default_output=format_output("Setting fan speed level to {fan_level}"),
+    )
+    def set_fan_level(self, fan_level):
+        """Set fan speed level (only for models with fan)"""
+        if self.model in MODELS_WITH_FAN_WY or self.model in MODELS_WITH_FAN_WY2:
+            if fan_level < 0 or fan_level > 100:
+                raise HuizuoException("Invalid fan speed level: %s" % fan_level)
+
+            return self.set_property("fan_level", fan_level)
+        else:
+            raise HuizuoException("Your device doesn't support a fan management")
+
+    @command(
+        default_output=format_output("Setting fan mode to 'Basic'"),
+    )
+    def set_basic_fan_mode(self):
+        """Set fan mode to 'Basic' (only for models with fan)"""
+        if self.model in MODELS_WITH_FAN_WY or self.model in MODELS_WITH_FAN_WY2:
+            return self.set_property("fan_mode", 0)
+        else:
+            raise HuizuoException("Your device doesn't support a fan management")
+
+    @command(
+        default_output=format_output("Setting fan mode to 'Natural wind'"),
+    )
+    def set_natural_fan_mode(self):
+        """Set fan mode to 'Natural wind' (only for models with fan)"""
+        if self.model in MODELS_WITH_FAN_WY or self.model in MODELS_WITH_FAN_WY2:
+            return self.set_property("fan_mode", 1)
+        else:
+            raise HuizuoException("Your device doesn't support a fan management")
+
+    @command(
+        default_output=format_output(
+            "\n",
+            "------------ Lamp parameters -----------\n"
+            "Power: {result.is_on}\n"
+            "Brightness: {result.brightness}\n"
+            "Color Temperature: {result.color_temp}\n"
+            "\n"
+            "------------Fan parameters -------------\n"
+            "Fan power:  {result.is_fan_on}\n"
+            "Fan level: {result.fan_speed_level}\n"
+            "Fan mode: {result.fan_mode}\n"
+            "Fan reverse: {result.is_fan_reverse}\n"
+            "\n",
+        ),
+    )
+    def status(self) -> HuizuoStatus:
+        """Retrieve properties."""
```

Comment:
```suggestion
        """Return status of the device."""
```
Looks nicer in `--help` and apidocs that way.

Reply:
Also, this probably should be in `Huizuo` class to avoid code duplication? See below.

Reply:
We have a different output format (list of available fields) in different classes. It depends on the available functions of the lamp. That's why I'm re-defining the 'HuizuoStatus' method in every class.
Or you have another suggestion (maybe I do not fully understand your idea)?

Reply:
Hmm, but the unavailable properties would then just be reported as `None` which would be fine (in my opinion). I'm not completely against having separate status outputs, but I'd rather have a single status response for maintainability.

Reply:
I tried this already in the version, which I tested internally with my basic lamps, and I made the method overloading for the next reasons:
1. I don't know the full model line of Huizuo lamps, I only added what I found in the Issue description. If they have/will have more devices with different new functions (like RGB support, Ambilight, etc.), this single list will be quite long, mostly filled with "none" for basic devices. Not very comfortable (IMO) for reading
2. Device can have the functions from different sections in the list, which may be separated (e.g. basic in the beginning and RGB at the end of the list).
Method overloading allows simplifying the reading of the properties of the specific device type.

But, if you have already a "single list" approach for other devices with similar classes organizations, indeed, better bring it to the single style.
Just inform me about the decision ;)

Reply:
Okay, let's have it like it is right now, we can always revisit that in the future if needed :-) Thanks again for the great PR! :tada: